app/operations/v1/padlock/sessions/create.rb

```diff
@@ -0,0 +1,42 @@
+module V1
+  module Padlock
+    module Sessions
+      class Create < ::Callable
+        def initialize(params = {})
+          @password = params[:password]
+          @email    = params[:email]
+          @phone    = params[:phone]
+        end
+
+        def call
+          return unless identity && user.authenticate(password)
+
+          generate_token
+
+          user
+        end
+
+        private
+
+        attr_reader :password, :email, :phone
+
+        def identity
+          @identity ||= case
+                        when email
+                          Identity.find_by(uid: email)
```

Comment:
```
def uid
  email || phone
end

def identity
  @identity ||= Identity.find_by(uid: uid)
end
```

Reply:
![5-levels-of-customer-satisfaction](https://user-images.githubusercontent.com/19553805/38600861-58168f76-3d6e-11e8-9ab1-b64a5bece5a8.jpg)